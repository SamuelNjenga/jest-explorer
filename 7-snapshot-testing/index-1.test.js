//Property Matchers
//Often there are fields in the object you want to snapshot which are generated (like IDs and Dates). If you try to snapshot these objects, they will force the snapshot to fail on every run
it('will fail every time', () => {
    const user = {
      createdAt: new Date(),
      id: Math.floor(Math.random() * 20),
      name: 'Javascript Engineer',
    };
  
    expect(user).toMatchSnapshot();
  });
  
  // Snapshot
//   exports[`will fail every time 1`] = `
//   Object {
//     "createdAt": 2018-05-19T23:36:09.816Z,
//     "id": 3,
//     "name": "LeBron James",
//   }
//   `;
