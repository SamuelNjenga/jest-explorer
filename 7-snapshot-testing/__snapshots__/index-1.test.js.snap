// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`will fail every time 1`] = `
Object {
  "createdAt": 2020-07-10T16:50:18.496Z,
  "id": 14,
  "name": "Javascript Engineer",
}
`;
